sycl/doc/extensions/proposed/sycl_ext_oneapi_enqueue_functions.asciidoc

```diff
@@ -0,0 +1,684 @@
+!====
+_Effects_: Submits a command-group function object (as defined by the SYCL
+specification) to the `sycl::queue`.
+
+a|
+[frame=all,grid=none]
+!====
+a!
+[source,c++]
+----
+namespace sycl::ext::oneapi::experimental {
+
+template <typename CommandGroupFunc>
+sycl::event submit_with_event(sycl::queue q, CommandGroupFunc&& cgf);
+
+}
+----
+!====
+_Effects_: Submits a command-group function object (as defined by the SYCL
+specification) to the `sycl::queue`.
+
+_Returns_: A `sycl::event` associated with the submitted command.
+
+|====
+
+
+=== Commands
+
+==== Single tasks
+
+|====
+a|
+[frame=all,grid=none]
+!====
+a!
+[source,c++]
+----
+namespace sycl::ext::oneapi::experimental {
+
+template <typename KernelName, typename KernelType>
+void single_task(sycl::queue q, const KernelType& k);
+
+template <typename KernelName, typename KernelType>
+void single_task(sycl::handler h, const KernelType& k);
+
+}
+----
+!====
+_Effects_: Enqueues a kernel function to the `sycl::queue` or `sycl::handler`
+as a single task.
+
+a|
+[frame=all,grid=none]
+!====
+a!
+[source,c++]
+----
+namespace sycl::ext::oneapi::experimental {
+
+template <typename Args...>
+void single_task(sycl::queue q, const sycl::kernel& k, Args&&... args);
+
+template <typename Args...>
+void single_task(sycl::handler h, const sycl::kernel& k, Args&&... args);
+
+}
+----
+!====
+_Effects_: Enqueues a kernel object to the `sycl::queue` or `sycl::handler` as
+a single task. The arguments in `args` are passed to the kernel in the same
+order.
+
+|====
+
+
+==== Basic kernels
+
+|====
+a|
+[frame=all,grid=none]
+!====
+a!
+[source,c++]
+----
+namespace sycl::ext::oneapi::experimental {
+
+template <typename KernelName, int Dimensions, typename KernelType, typename... Reductions>
+void parallel_for(sycl::queue q, sycl::range<Dimensions> r, const KernelType& k, Reductions&&... reductions);
+
+template <typename KernelName, int Dimensions, typename KernelType, typename... Reductions>
+void parallel_for(sycl::handler h, sycl::range<Dimensions> r, const KernelType& k, Reductions&&... reductions);
+
+}
+----
+!====
+_Constraints_: The parameter pack consists of 0 or more objects created by the
+`sycl::reduction` function.
+
+_Effects_: Enqueues a kernel function to the `sycl::queue` or `sycl::handler`
+as a basic kernel, using the number of work-items specified by a `sycl::range`.
+
+a|
+[frame=all,grid=none]
+!====
+a!
+[source,c++]
+----
+namespace sycl::ext::oneapi::experimental {
+
+template <typename KernelName, int Dimensions, typename Properties, typename KernelType, typename... Reductions>
+void parallel_for(sycl::queue q, launch_config<sycl::range<Dimensions>, Properties> c, const KernelType& k, Reductions&&... reductions);
+
+template <typename KernelName, int Dimensions, typename Properties, typename KernelType, typename... Reductions>
+void parallel_for(sycl::handler h, launch_config<sycl::range<Dimensions>, Properties> c, const KernelType& k, Reductions&&... reductions);
```

Comment:
These function signatures are pretty long, causing the table to have a horizontal scroll bar on my screen.

Reply:
I tried to fix in 656ddebb8090d8221bbf4716f974a4bc5e695fc0.

The old format didn't suffer as much from this (because the scrollbar only applied to the code box, not to the whole definition).  I think this new format is going to require us to wrap definitions more aggressively, so we might need to come up with some style guide for where we put the line breaks.

Reply:
One of the scrollbars is still there, but I can't figure out which line is too long.  As far as I can tell, they're all wrapped at 80 columns.

Reply:
Ok, the combination of https://github.com/intel/llvm/pull/11491/commits/2b8ffc8da9886d9b75135bdf5b78a87f23c108e6 and https://github.com/intel/llvm/pull/11491/commits/24b3a217d0a48f84c12fcb68c04e58fc156409d6 has fixed the scrollbar issue.